frontend/src/components/MapView/LeftPanel/layersPanel/MenuSwitch/SwitchItem/ExposureAnalysisOption.tsx

```diff
@@ -0,0 +1,97 @@
+import { IconButton, Tooltip } from '@material-ui/core';
+import { ImageAspectRatioOutlined } from '@material-ui/icons';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  AggregationOperations,
+  ExposedPopulationDefinition,
+  GeometryType,
+  LayerType,
+} from '../../../../../../config/types';
+import { TableKey } from '../../../../../../config/utils';
+import {
+  analysisResultSelector,
+  clearAnalysisResult,
+  ExposedPopulationDispatchParams,
+  requestAndStoreExposedPopulation,
+  setCurrentDataDefinition,
+} from '../../../../../../context/analysisResultStateSlice';
+import { setTabValue } from '../../../../../../context/leftPanelStateSlice';
+import { dateRangeSelector } from '../../../../../../context/mapStateSlice/selectors';
+import { useSafeTranslation } from '../../../../../../i18n';
+import { Extent } from '../../../../Layers/raster-utils';
+
+function ExposureAnalysisOption({
+  layer,
+  extent,
+  selected,
+  exposure,
+}: ExposureAnalysisOptionProps) {
+  const dispatch = useDispatch();
+  const { t } = useSafeTranslation();
+  const analysisResult = useSelector(analysisResultSelector);
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
+  // Since the exposure analysis doesn't have predefined table in configurations
+  // and need to have a `TableKey` will use this util function to handle such case
+  // used timestamp to avoid any potential rare name collision
+  const generateUniqueTableKey = (activityName: string) => {
+    return `${activityName}_${Date.now()}`;
+  };
```

Comment:
Since this function is duplicated, let's make it a util function to generate a unique key.